backends/ebpf/psa/ebpfPsaArch.cpp

```diff
@@ -0,0 +1,637 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "ebpfPsaArch.h"
+#include "ebpfPsaParser.h"
+#include "ebpfPsaDeparser.h"
+#include "ebpfPsaTable.h"
+#include "ebpfPsaControl.h"
+#include "xdpHelpProgram.h"
+
+namespace EBPF {
+
+// =====================PSAArch=============================
+void PSAArch::emitPSAIncludes(CodeBuilder *builder) const {
+    builder->appendLine("#include <stdbool.h>");
+    builder->appendLine("#include <linux/if_ether.h>");
+    builder->appendLine("#include \"psa.h\"");
+}
+
+void PSAArch::emitPreamble(CodeBuilder *builder) const {
+    emitCommonPreamble(builder);
+    builder->newline();
+
+    builder->appendLine("#define CLONE_MAX_PORTS 64");
```

Comment:
This kind of stuff could also be in a fixed header file. But perhaps it's better to have fewer of these hanging around?

Reply:
Exactly. This could probably be configurable with compiler options in future. I'll add a TODO.